### ISSUE TITLE
seeking a working combination of schmitty, agda, agda-stdlib and agdarsec

### ISSUE THREAD

**Issue by jmchapman** [wenkokke/schmitty#28]
I tried to install the lower bounds of the suggestions in the README:
Agda (>= 2.6.2-60564a4)
agda-stdlib (>= experimental-d66a21f)
agdarsec (>= master-b26230)

But I get this error:

agda-stdlib/src/Data/List/Membership/Propositional/Properties.agda:17,1-52
Failed to solve the following constraints:
  Check definition of to∘from : {P.A.a : Level} {P.A : Set P.A.a}
                                {P.p : Level} {P = P₁ : Pred P.A P.p} {Q.A.a : Level}
                                {Q.A : Set Q.A.a} {Q.p : Level} {Q = Q₁ : Pred Q.A Q.p}
                                {xs = xs₁ : List P.A} {ys = ys₁ : List Q.A}
                                (pq
                                 : Any (λ x → Any (λ y → Prod.Σ (P₁ x) (λ x₁ → Q₁ y)) ys₁) xs₁) →
                                Any-×⁺ (Any-×⁻ pq) ≡ pq
    stuck because
      agda-stdlib/src/Data/List/Relation/Unary/Any/Properties.agda:266,3-34
      I'm not sure if there should be a case for the constructor refl,
      because I get stuck when trying to solve the following unification
      problems (inferred index ≟ expected index):
        lhs ≟ Any.map
              (λ x →
                 Any.map (λ q → P.subst P x p , q)
                 (Any.map
                  (λ x₁ →
                     P.subst Q x₁
                     (_5428 (x = x₂) (pq′ = pq′) (y = y) (y∈ys = y∈ys) (p = p) (q = q)
                      (lem₂ = lem₂) (pq = pq) (x∈xs = x∈xs) (lem₁ = lem₁)
                      (p = (P.subst P x p))))
                  y∈ys))
              x∈xs
      when checking that the pattern refl has type
      lhs ≡
      Any.map
      (λ x →
         Any.map (λ q → P.subst P x p , q)
         (Any.map
          (λ x₁ →
             P.subst Q x₁
             (_5428 (x = x₂) (pq′ = pq′) (y = y) (y∈ys = y∈ys) (p = p) (q = q)
              (lem₂ = lem₂) (pq = pq) (x∈xs = x∈xs) (lem₁ = lem₁)
              (p = (P.subst P x p))))
          y∈ys))
      x∈xs
    (blocked on _P.p_475)
when scope checking the declaration
  open import Data.List.Relation.Unary.Any.Properties
agda-stdlib/src/Data/List/Membership/Propositional/Properties.agda:17,1-52
Unsolved metas at the following locations:
  agda-stdlib/src/Data/List/Relation/Unary/Any/Properties.agda:100,17-18
  agda-stdlib/src/Data/List/Relation/Unary/Any/Properties.agda:105,16-17
  agda-stdlib/src/Data/List/Relation/Unary/Any/Properties.agda:105,28-29
when scope checking the declaration
  open import Data.List.Relation.Unary.Any.Properties

I also tried the upper bounds - i.e., the latest versions

I get this error:

schmitty/src/Text/Parser/String.agda:53,17-31
Not in scope:
  Parser.Success
  at schmitty/src/Text/Parser/String.agda:53,17-31
    (did you mean 'Success'?)
when scope checking Parser.Success

**Comment by MatthewDaggitt** [comment]
Err the first error appears to be the same reported in https://github.com/agda/agda-stdlib/issues/1414. I suspect that means that either your dev agda standard library experimental branch or your dev agda version isn't as far along as you think it is?

I'm very hopeful that once Agda 2.6.2 releases, these stability issues are going to go away :crossed_fingers:

**Comment by jmchapman** [comment]
Thanks @MatthewDaggitt.

I hope this is the case but I am still confused:

$ agda --version
Agda version 2.6.2-60564a4

$ cd agda-stdlib
$ git status
HEAD detached at d66a21f1

$ cd
$ cat schmitty.agda
module schmitty where

{-# OPTIONS --allow-exec #-}

open import Data.Integer
open import Data.List
open import Data.Product
open import Relation.Binary.PropositionalEquality
open import SMT.Theories.Ints as Ints
open import SMT.Backend.Z3 Ints.reflectable
$ agda schmitty.agda
(same error as before)

I can't think what else I can be doing wrong?

**Comment by MatthewDaggitt** [comment]
With the following your code works fine for me:

- agda/agda@60564a4
- agda/agda-stdlib@d66a21f
- gallais/agdarsec@b26230a 
- wenkokke/schmitty@00585a3

If you're getting the `Parser.Success` error then maybe your `agdarsec` version is wrong (only thing you didn't supply in the code above)? If you're getting the unification error, then I'm afraid I'm at a complete loss.

**Comment by jmchapman** [comment]
@MatthewDaggitt thanks for checking. I'll have another go.

**Comment by wenkokke** [comment]
@jmchapman With the release of agda v2.6.2 we now have a stable version for schmitty!

**Comment by jmchapman** [comment]
Great!
